internal/lang/functions.go

```diff
@@ -189,12 +189,22 @@ func (s *Scope) Functions() map[string]function.Function {
 			}
 		}
 
+		coreNames := make([]string, 0)
 		// Add a description to each function and parameter based on the
 		// contents of descriptionList.
 		// One must create a matching description entry whenever a new
 		// function is introduced.
 		for name, f := range s.funcs {
 			s.funcs[name] = funcs.WithDescription(name, f)
+			coreNames = append(coreNames, name)
+		}
+		// Copy all stdlib funcs into core:: namespace
+		for _, name := range coreNames {
+			s.funcs["core::"+name] = s.funcs[name]
```

Comment:
I don't understand why we populate the non-provider functions twice, one with `core::` and once without. Can you please explain?

Reply:
Terraform set that convention, I honestly don't know why they did as I have tried to look at their documentation as little as possible when working on this feature.  I just know that I wanted it to be as compatible as possible.